src/Microsoft.ML/CSharpApi.cs

```diff
@@ -7599,6 +7599,14 @@ public sealed partial class LightGbmBinaryClassifier : Microsoft.ML.Runtime.Entr
             [TlcModule.SweepableDiscreteParamAttribute("CatL2", new object[]{0.1f, 0.5f, 1, 5, 10})]
             public double CatL2 { get; set; } = 10d;
 
+            /// <summary>
+            /// Execution device for training (CPU or GPU).
+            /// NOTE: GPU training requires compatible build of LightGBM as described here:
+            /// https://github.com/Microsoft/LightGBM/blob/master/docs/Installation-Guide.rst#build-gpu-version
+            /// </summary>
```

Comment:
might be worth moving this comment on the actual LightGBMArguments file, since this file is generated. Mabye on the ExecutionDevice classes.